Add the Repositories and repair the content for RHEL 8

mjivraja opened the pull request:
Added the new repositories for Satellite and Capsule Servers v6.11
supporting RHEL 8. Also, added the new content and modified the
existing one wherever required to make the documentation more accurate.

Support Satellite and Capsule running on RHEL 7 and RHEL 8 - commands
in the installation docs should reflect RHEL 8.

https://issues.redhat.com/browse/SAT-9421


Cherry-pick into:

* [X] Foreman 3.3
* [X] Foreman 3.2
* [X] Foreman 3.1
* For Foreman 3.0 or older, file a separate PR request

<!---
Thank you for contributing to Foreman documentation. Make sure to read README
for the documentation standards. Set cherry-pick github label to mark this
contribution for cherry picking and check which version do you need with [x].
-->

maximiliankolb commented:
Hi @mjivraja Before we can review your PR, please rebase it against master. Also think about our conventions and the use of hardcoded values such as "capsule". You can most likely use attributes instead.

mjivraja commented:
> Hi @mjivraja Before we can review your PR, please rebase it against master. Also think about our conventions and the use of hardcoded values such as "capsule". You can most likely use attributes instead.

Hi @maximiliankolb 

Master is showing up-to-date while trying to rebase. 

Do you mean there should be an individual attribute for the word 'Capsule'? I am sorry if I didn't understand correctly as I am still learning the standard writing process of Satellite/Foreman in AsciiDoc.

maximiliankolb commented:
Hey @mjivraja 
Github tells me that your commits are **not** on top of master, but an earlier commit in master.

Here's how to proof that:

```
$ cd foreman-documentation
$ git switch master
$ git remote add mjivraja git@github.com:mjivraja/foreman-documentation.git
$ git fetch --all --prune --tags
$ git pull --all
$ git merge-base upstream/master mjivraja/SAT-9421-Support-Satellite-and-Capsule-running-on-RHEL-7-and-RHEL-8
```

This returns a commit ID that is **not** HEAD of master but another commit, "deep down", in master.

You most likely need to fetch and pull from https://github.com/theforeman/foreman-documentation/; the exact commands depend on your remotes:

```
$ git remote -v
github	git@github.com:ATIX-AG/foreman-documentation.git (fetch)
github	git@github.com:ATIX-AG/foreman-documentation.git (push)
mjivraja	git@github.com:mjivraja/foreman-documentation.git (fetch)
mjivraja	git@github.com:mjivraja/foreman-documentation.git (push)
origin	ssh://git@git.atix.de:_REDACTED_foreman-documentation.git (fetch)
origin	ssh://git@git.atix.de:_REDACTED_foreman-documentation.git (push)
upstream	git@github.com:theforeman/foreman-documentation.git (fetch)
upstream	git@github.com:theforeman/foreman-documentation.git (push)
```

In this case, you'd want to pull from "upstream/master" into "master" in your local git repo. Afterwards, you can switch to your branch and do an interactive rebase.

maximiliankolb commented:
Maybe your local master branch it tracking your own remote master branch, i.e. the master branch in your github fork, but not the master branch of foreman-documentation in `theforeman` on github. Run `git remote -v` to view all remotes. You should have at least two: your username on github and theforeman on github. Then ensure you pull from the "master" branch from the origin that points to `github.com/theforeman/foreman-documentation`.

evgeni commented on guides/common/modules/proc_attaching-satellite-infrastructure-subscription.adoc:
```diff
@@ -2,8 +2,7 @@
 After you have registered {ProductName}, you must identify your subscription Pool ID and attach an available subscription.
-The {ProjectName} Infrastructure subscription provides access to the {ProjectName}, {RHEL}, and Red{nbsp}Hat Software Collections (RHSCL) content.
-This is the only subscription required.
+The {ProjectName} Infrastructure subscription provides access to the {ProjectName} and {RHEL} content. For {RHEL} 7, you may require Red Hat Software Collections (RHSCL). This is the only subscription required.
```
```suggestion
The {ProjectName} Infrastructure subscription provides access to the {ProjectName} and {RHEL} content. For {RHEL} 7 it also provides access to Red{nbsp}Hat Software Collections (RHSCL). This is the only subscription required.
```

mjivraja replied:
Modified.

evgeni commented on guides/common/modules/proc_configuring-capsule-repositories.adoc:
```diff
@@ -0,0 +1,168 @@
+* xref:#repositories-ubuntu-2004[Ubuntu 20.04 (Focal)]
+endif::[]
+
+ifdef::foreman-el,katello[]
+== [[repositories-centos-8]]CentOS 8
```
can you please re-sync this section with the latest version of `guides/common/modules/proc_configuring-repositories.adoc`, especially the changes made in ef2a39ee80b3438a38fe8dadc9ba9a97dfc108b1?

evgeni commented on guides/common/assembly_installing-capsule-server.adoc:
```diff
@@ -19,7 +19,7 @@ endif::[]
 ifdef::foreman-el,foreman-deb,satellite[]
 // Configuring Repositories
-include::modules/proc_configuring-repositories.adoc[leveloffset=+1]
+include::modules/proc_configuring-capsule-repositories.adoc[leveloffset=+1]
```
I wonder if this is a good idea, or not.
While Satellite does have different repositories for Satellite and Capsule, upstream does not.
So maybe `proc_configuring-capsule-repositories.adoc` should be Satellite-specific?

What does @melcorr think?

melcorr replied:
It's a very bad idea to add this. We really went to a lot of effort to remove all this.

melcorr replied:
> I wonder if this is a good idea, or not. While Satellite does have different repositories for Satellite and Capsule, upstream does not. So maybe `proc_configuring-capsule-repositories.adoc` should be Satellite-specific?
> 
Agreed. 

@mjivraja please, never ever ever add Satellite product names to file names. This is a single source that is used in the community and in other companies. We must remain neutral.

mjivraja replied:
Resolved. Added the Capsule repos in `proc_configuring-repositories-proxy.adoc`

evgeni commented on guides/common/attributes-base.adoc:
```diff
@@ -16,16 +16,26 @@
 :RepoRHEL7ServerOptional: rhel-7-server-optional-rpms
 :RepoRHEL7ServerAnsible: rhel-7-server-ansible-2.9-rpms
+// For RHEL 8
+:RepoRHEL8Server: rhel-8-for-x86_64-baseos-rpms
+:RepoRHEL8ServerSoftwareCollections: rhel-8-for-x86_64-appstream-rpms
```
AppStream is not the same thing as SoftwareCollections.

These should really be `RepoRHEL8Base` and `RepoRHEL8AppStream`

mjivraja replied:
Resolved.

evgeni commented on guides/common/attributes-base.adoc:
```diff
@@ -16,16 +16,26 @@
+// For RHEL 8
+:RepoRHEL8Server: rhel-8-for-x86_64-baseos-rpms
+:RepoRHEL8ServerSoftwareCollections: rhel-8-for-x86_64-appstream-rpms
+:RepoRHEL8ServerAnsible: ansible-2.9-for-rhel-8-x86_64-rpms
```
this is not needed anymore with the release of RHEL 8.6

evgeni reviewed (commented):
not a full review, but a few comments

melcorr commented:
Hey @mjivraja thanks for this PR. 
As well as everything @maximiliankolb has pointed to about our conventions, please do not use the terms Satellite and Capsule in the PR description. I know that it can be difficult to make this mind-shift, but this is an open source project, in the upstream, and these Red Hat product terms should not feature anywhere here.

mjivraja commented:
> Maybe your local master branch it tracking your own remote master branch, i.e. the master branch in your github fork, but not the master branch of foreman-documentation in `theforeman` on github. Run `git remote -v` to view all remotes. You should have at least two: your username on github and theforeman on github. Then ensure you pull from the "master" branch from the origin that points to `github.com/theforeman/foreman-documentation`.

`git remote -v` returned:
```
mjivraja	git@github.com:mjivraja/foreman-documentation.git (fetch)
mjivraja	git@github.com:mjivraja/foreman-documentation.git (push)
origin	git@github.com:mjivraja/foreman-documentation.git (fetch)
origin	git@github.com:mjivraja/foreman-documentation.git (push)
upstream	git@github.com:theforeman/foreman-documentation.git (fetch)
upstream	git@github.com:theforeman/foreman-documentation.git (push)
```

`git remote get-url origin` returned:
```
git@github.com:mjivraja/foreman-documentation.git
```
I think, what the last command returned is not correct as per your explanation. It should be `github.com/theforeman/foreman-documentation`

mjivraja commented:
> Hey @mjivraja thanks for this PR. As well as everything @maximiliankolb has pointed to about our conventions, please do not use the terms Satellite and Capsule in the PR description. I know that it can be difficult to make this mind-shift, but this is an open source project, in the upstream, and these Red Hat product terms should not feature anywhere here.

Noted @melcorr  I understood better. I will be careful from next time and if required,  I can try to edit the commit message.

maximiliankolb commented:
> > Maybe your local master branch it tracking your own remote master branch, i.e. the master branch in your github fork, but not the master branch of foreman-documentation in `theforeman` on github. Run `git remote -v` to view all remotes. You should have at least two: your username on github and theforeman on github. Then ensure you pull from the "master" branch from the origin that points to `github.com/theforeman/foreman-documentation`.
> 
> `git remote -v` returned:
> 
> ```
> mjivraja	git@github.com:mjivraja/foreman-documentation.git (fetch)
> mjivraja"	git@github.com:mjivraja/foreman-documentation.git (push)
> origin	git@github.com:mjivraja/foreman-documentation.git (fetch)
> origin	git@github.com:mjivraja/foreman-documentation.git (push)
> upstream	git@github.com:theforeman/foreman-documentation.git (fetch)
> upstream	git@github.com:theforeman/foreman-documentation.git (push)
> ```
> 
> `git remote get-url origin` returned:
> 
> ```
> git@github.com:mjivraja/foreman-documentation.git
> ```
> 
> I think, what the last command returned is not correct as per your explanation. It should be `github.com/theforeman/foreman-documentation`

I assume you **DO NOT** have a local branch master that tracks upstream/master, but you should. Run `git switch master && git status` to see which remote (either mjivraja, origin, or upstream) your branch tracks. I suspects it's not "upstream", but it should be. You can use `-u` (see https://stackoverflow.com/a/2286030) to change the remote that it tracks. After that, a normal fetch and pull should suffice. Then your rebase should work as expected.

Also, you have your fork on github twice in there; once as "origin" and once as "mjivraja", which is not necessary.

melcorr commented:
Thanks  @maximiliankolb - I think we are on the right track now and he is working on fixing all conflicts and rebasing!

melcorr commented on guides/common/attributes-base.adoc:
```diff
@@ -39,6 +39,7 @@
 // Do not update the puppet4 repo versions. They must stay at 6.3.
 :RepoRHEL7ServerSatelliteServerPuppetVersion: rhel-7-server-satellite-6.3-puppet4-rpms
 :RepoRHEL7ServerSatelliteCapsulePuppetVersion: rhel-7-server-satellite-capsule-6.3-puppet4-rpms
+>>>>>>> da40753a (Add the Repositories and repair the content for RHEL 8)
```
please remove this....

mjivraja replied:
Resolved

melcorr commented on guides/common/modules/proc_configuring-repositories.adoc:
```diff
@@ -11,6 +11,7 @@ endif::[]
 ifdef::foreman-el,katello,satellite[]
 * xref:#repositories-rhel-8[Red Hat Enterprise Linux 8]
 * xref:#repositories-rhel-7[Red Hat Enterprise Linux 7]
+>>>>>>> da40753a (Add the Repositories and repair the content for RHEL 8)
```
something went wrong with fixing conflicts...

mjivraja replied:
Yes. The Atom might be. Resolved.

maximiliankolb commented:
> Please accept my apology for this action. This wasn't intended at all. My mouse went jittery and clicked the Push button in Atom. I will be active to fix this in priority.
> 
> @melcorr @maximiliankolb

no worries mate. ping if if it's ready for review.

mjivraja commented:
Hi @maximiliankolb 

Thank you for your help as finally, all the checks have passed for my latest commit today. I would like to request you to review my changes and let me know of any modifications required further.

Thank you.

mjivraja commented:
Hi @evgeni 

Finally, I have made the documentation more accurate and resolved some complex errors, I would like to request you to review the changes from your side once again. I have addressed all the previous recommendations from your side and am willing to revise more if required.

Looking forward to your review.

Thank you.

melcorr commented on guides/common/modules/proc_configuring-repositories-proxy.adoc:
```diff
@@ -7,8 +7,54 @@ This procedure is only for Katello plug-in and {RHEL}-based operating system use
+# dnf module enable satellite-capsule:el8
+----
++
+endif::[]
```
delete one of these also

mjivraja replied:
Deleted

melcorr commented on guides/common/modules/proc_configuring-repositories.adoc:
```diff
@@ -70,6 +71,43 @@ include::proc_configuring-repositories-el.adoc[]
++
+[options="nowrap"]
+----
+# dnf module enable satellite:el8
```
I can't imagine that this is true...

evgeni replied:
and yet it is :)

melcorr reviewed (commented):
Few comments from me but this needs a bigger review

evgeni commented on guides/common/modules/proc_configuring-repositories-proxy.adoc:
```diff
@@ -7,8 +7,54 @@ This procedure is only for Katello plug-in and {RHEL}-based operating system use
++
+[options="nowrap"]
+----
+# dnf module enable satellite-capsule:el8
```
On EL8, sure.

evgeni commented on guides/common/attributes-base.adoc:
```diff
@@ -18,12 +18,24 @@
 :RepoRHEL7ServerOptional: rhel-7-server-optional-rpms
 :RepoRHEL7ServerAnsible: rhel-7-server-ansible-2.9-rpms
-:RepoRHEL8ServerAnsible: ansible-2.9-for-rhel-8-x86_64-rpms 
+// For Beta, change to "Beta". For GA releases, change to, for example, "6.8".
+:RepoRHEL7ServerSatelliteServerProductVersion: rhel-7-server-satellite-6.11-rpms
+:RepoRHEL7ServerSatelliteServerProductVersionPrevious: rhel-server-7-satellite-6.9-rpms
+:RepoRHEL7ServerSatelliteCapsuleProductVersion: rhel-7-server-satellite-capsule-6.11-rpms
+:RepoRHEL7ServerSatelliteToolsProductVersion: rhel-7-server-satellite-tools-6.10-rpms
+:RepoRHEL7ServerSatelliteMaintenanceProductVersion: rhel-7-server-satellite-maintenance-6.11-rpms
+// For RHEL 8
+:RepoRHEL8ServerSatelliteServerProductVersion: satellite-6.11-for-rhel-8-x86_64-rpms
+:RepoRHEL8ServerSatelliteMaintenanceProductVersion: satellite-maintenance-6.11-for-rhel-8-x86_64-rpms
+:RepoRHEL8ServerSatelliteCapsuleProductVersion: satellite-capsule-6.11-for-rhel-8-x86_64-rpms
+// Do not update the puppet4 repo versions. They must stay at 6.3.
+:RepoRHEL7ServerSatelliteServerPuppetVersion: rhel-7-server-satellite-6.3-puppet4-rpms
+:RepoRHEL7ServerSatelliteCapsulePuppetVersion: rhel-7-server-satellite-capsule-6.3-puppet4-rpms
```
I think all these should go into `guides/common/attributes-satellite.adoc`, not into `guides/common/attributes-base.adoc`

evgeni replied:
actually, they already are?

https://github.com/theforeman/foreman-documentation/blob/e18c3c19b7a5472daa7c3c69a8bea1c70e5d32eb/guides/common/attributes-satellite.adoc#L101-L110

mjivraja replied:
Yes. Those attributes are covered in `attributes-satellite`. Removed from `attribute-base'. Thank you for this.

evgeni commented on guides/common/modules/proc_configuring-repositories-proxy.adoc:
```diff
@@ -7,8 +7,54 @@ This procedure is only for Katello plug-in and {RHEL}-based operating system use
+== [[repositories-rhel-8]]Red Hat Enterprise Linux 8
+
+:distribution-major-version: 8
+:package-manager: yum, dnf
```
Yeah, it should be just `dnf` here.

mjivraja replied:
Removed Yum. There is only DNF now.

evgeni commented on guides/common/modules/proc_configuring-repositories.adoc:
```diff
@@ -99,20 +137,18 @@ ifdef::foreman-el,katello[]
+--enable={RepoRHEL7ServerSatelliteMaintenanceProductVersion}
 ----
 endif::[]
-ifdef::foreman-el,katello[]
-include::proc_configuring-repositories-el.adoc[]
-endif::[]
```
why are you removing this? it is still relevant on non-Satellite.

mjivraja replied:
It was extra and caused checks failure. Foreman and Katello will not face any issues as it is covered for both.

evgeni commented on guides/common/modules/ref_supported-operating-systems.adoc:
```diff
@@ -3,8 +3,8 @@
 You can install the operating system from a disc, local ISO image, kickstart, or any other method that Red{nbsp}Hat supports.
-Red{nbsp}Hat {ProductName} is supported only on the latest versions of {RHEL} 7 Server that is available at the time when {ProductName} is installed.
-Previous versions of {RHEL} including EUS or z-stream are not supported.
+Red{nbsp}Hat {ProductName} is supported on the latest versions of {RHEL} 8 Server, and {RHEL} 7 Server that are available at the time when {ProductName} is installed.
```
```suggestion
Red{nbsp}Hat {ProductName} is supported on the latest versions of {RHEL} 8, and {RHEL} 7 Server that are available at the time when {ProductName} is installed.
```

RHEL8 has no difference between Server and Workstation anymore

mjivraja replied:
Resolved.

evgeni commented on guides/common/modules/ref_supported-operating-systems.adoc:
```diff
@@ -3,8 +3,8 @@
-Red{nbsp}Hat {ProductName} is supported only on the latest versions of {RHEL} 7 Server that is available at the time when {ProductName} is installed.
-Previous versions of {RHEL} including EUS or z-stream are not supported.
+Red{nbsp}Hat {ProductName} is supported on the latest versions of {RHEL} 8 Server, and {RHEL} 7 Server that are available at the time when {ProductName} is installed.
+Previous versions of Red{nbsp}Hat Enterprise Linux including EUS or z-stream are not supported.
```
```suggestion
Previous versions of {RHEL} including EUS or z-stream are not supported.
```

mjivraja replied:
Resolved.

mjivraja commented:
There are many commits from my peers as I merged and rebased the branch. I hope this is normal.

maximiliankolb commented:
Please also address and resolve all open threads so that a new reviewer does not have to look over all improvements/changes that have already been made.

mjivraja commented:
> Please also address and resolve all open threads so that a new reviewer does not have to look over all improvements/changes that have already been made.

Apologies as I encountered a sudden meeting with my manager. I am addressing all the comments.

mjivraja commented:
> Something looks odd here. I think you need to rebase to master again. 126 files sounds like a lot & I think there are some changes from Lena in there. Thanks.

Ok. I am rebasing the master and see if there are any conflicts.

mjivraja commented on guides/common/modules/proc_configuring-repositories-proxy.adoc:
```diff
@@ -7,8 +7,54 @@ This procedure is only for Katello plug-in and {RHEL}-based operating system use
 
-.Procedure
+ifdef::foreman-el,katello,satellite[]
+* xref:#repositories-rhel-8[Red Hat Enterprise Linux 8]
+* xref:#repositories-rhel-7[Red Hat Enterprise Linux 7]
```
Resolved.

mjivraja commented on guides/common/modules/proc_registering-to-red-hat-subscription-management.adoc:
```diff
@@ -6,7 +6,8 @@ If you use a {RHEL}-based operating system, complete the following steps.
 
 Registering the host to Red Hat Subscription Management enables the host to subscribe to and consume content for any subscriptions available to the user.
-This includes content such as {RHEL}, Red Hat Software Collections (RHSCL), and {ProjectName}.
+This includes content such as Red Hat Enterprise Linux and {ProjectName}.
```
Resolved.

evgeni commented on guides/common/assembly_installing-capsule-server.adoc:
```diff
@@ -19,7 +19,19 @@ endif::[]
 ifdef::foreman-el,foreman-deb,satellite[]
 // Configuring Repositories
-include::modules/proc_configuring-repositories.adoc[leveloffset=+1]
+<<<<<<< HEAD
```
please rebase this on top of `master`

maximiliankolb commented on guides/common/modules/proc_installing-the-satellite-server-packages.adoc:
```diff
@@ -32,7 +32,6 @@ include::proc_installing-server-packages-el.adoc[]
 endif::[]
 
 ifdef::foreman-el,katello,satellite[]
-== [[rhel-7]]{RHEL} 7
```
Wouldn't you instead remove satellite from the line above and leave this line as is?

mjivraja replied:
Ohh yes. I was occupied more with git mess. I will be considering it.

maximiliankolb replied:
You did NOT change this.

mjivraja replied:
I am waiting for some time in case there are comments from other reviewers. If it's done, I am ready to make changes and push.

mjivraja replied:
Resolved

evgeni replied:
This wasn't resolved. The Satellite docs now render with an empty "Installing packages" section:
![image](https://user-images.githubusercontent.com/94284/172339110-4fbd159e-3d73-4c7c-9344-5d717b1041f0.png)

Moreso, this document didn't gain a RHEL8 section at all?!

maximiliankolb replied:
@mjivraja

maximiliankolb reviewed (commented):
One small comment.

mjivraja commented:
HI @maximiliankolb @evgeni @melcorr 

I have addressed all your comments in this single commit (except the latest from Maximillian). If there are any other reviews, please mention them. I will be taking it as a priority as it is now a blocker for one of the processes waiting for this ticket to get closed.

Sincere apologies for the Git mess.

Thank you.

maximiliankolb reviewed (approved):
LGTM.

maximiliankolb commented:
Handing over to @Lennonka for a final review & merge/cherry-pick.

Lennonka commented:
@evgeni Sorry, mistake.

evgeni commented on guides/common/modules/proc_configuring-repositories.adoc:
```diff
@@ -70,6 +71,43 @@ include::proc_configuring-repositories-el.adoc[]
+== [[repositories-rhel-8]]{RHEL} 8
+
+:distribution-major-version: 8
+:package-manager: yum, dnf
```
```suggestion
:package-manager: dnf
```

mjivraja replied:
Resolved

Lennonka commented on guides/common/modules/proc_configuring-repositories-proxy.adoc:
```diff
@@ -7,8 +7,54 @@ This procedure is only for Katello plug-in and {RHEL}-based operating system use
+endif::[]
+
+. Enable the following repositories:
+ifdef::satellite[]
```
It won't, change it here and now, I'll get to the review next week.

mjivraja replied:
> It won't, change it here and now, I'll get to the review next week.

@maximiliankolb  @melcorr  I don't have repos for non-satellite EL8. Can anyone please share it with me. Evgeni is out of office till 7 June. Need to close this PR at the earliest.

melcorr replied:
@ehelms any chance you could help with this?

Lennonka replied:
Okay, in that case it might be a good idea to split it.

mjivraja replied:
Thank you @Lennonka 

@maximiliankolb @melcorr Can we proceed further if there are no further reviews?

maximiliankolb replied:
I would have loved to see this resolved with this PR. But we can still merge as is if you create an issue on Github so you/we don't forget about this.

I made one small suggestion and also closed all your resolved threads as part of my re-re-review. Tip: Please comment on open threads in the "review mode" on the code tab and then close all resolved threads so that future reviewers don't have to re-read all your earlier stages.

Please link the issue & I will merge.

mjivraja replied:
> made one small suggestion and also closed all your resolved threads as part of my re-re-review.

I have staged it and ready to push.

> But we can still merge as is if you create an issue on Github so you/we don't forget about this.

I will raise the issue here on Github post this PR. As soon as I get the repos from Evgeni or Eric, I will process the same.

> Tip: Please comment on open threads in the "review mode" on the code tab and then close all resolved threads so that future reviewers don't have to re-read all your earlier stages.

Once it is pushed, I am commenting on the review and mark it as Resolved in the 'Conversation' tab. The same reflects on the 'Files Changed' tab. I hope I understand your tip in the correct manner.

maximiliankolb commented on guides/common/modules/proc_configuring-repositories.adoc:
```diff
@@ -70,6 +71,43 @@ include::proc_configuring-repositories-el.adoc[]
+--enable={RepoRHEL8ServerSatelliteServerProductVersion} \
+--enable={RepoRHEL8ServerSatelliteMaintenanceProductVersion}
+----
+endif::[]
+ifdef::satellite[]
```
```suggestion
```

Both steps are satellite-only.

mjivraja replied:
Addressed.

maximiliankolb commented:
Merged to master and cherry-picked ✅ :
   c2b9ed5f..b2866cae  3.1 -> 3.1
   56f4ba42..13c069dd  3.2 -> 3.2
   95ef2415..10f16857  3.3 -> 3.3
